include/mqtt/endpoint.hpp

```diff
@@ -8278,1217 +8325,3808 @@ class endpoint : public std::enable_shared_from_this<endpoint<Socket, Mutex, Loc
+                    );
+                    process_property_id(
+                        std::move(func),
+                        std::move(buf),
+                        rest,
+                        std::move(props),
+                        std::move(handler),
+                        std::move(self)
+                    );
+                },
+                std::move(self)
+            );
+        } break;
+        case v5::property::id::subscription_identifier_available: {
+            std::size_t const len = 1;
+            if (property_length_rest < len) {
+                call_message_size_error_handlers(func);
+                return;
+            }
+            process_nbytes(
+                std::move(func),
+                std::move(buf),
+                len,
+                [
+                    this,
+                    props = std::move(props),
+                    handler = std::move(handler),
+                    rest = property_length_rest - len
+                ]
+                (buffer body, buffer buf, async_handler_t func, this_type_sp self) mutable {
+                    props.emplace_back(
+                        v5::property::subscription_identifier_available(body.begin(), body.end())
+                    );
+                    process_property_id(
+                        std::move(func),
+                        std::move(buf),
+                        rest,
+                        std::move(props),
+                        std::move(handler),
+                        std::move(self)
+                    );
+                },
+                std::move(self)
+            );
+        } break;
+        case v5::property::id::shared_subscription_available: {
+            std::size_t const len = 1;
+            if (property_length_rest < len) {
+                call_message_size_error_handlers(func);
+                return;
+            }
+            process_nbytes(
+                std::move(func),
+                std::move(buf),
+                len,
+                [
+                    this,
+                    props = std::move(props),
+                    handler = std::move(handler),
+                    rest = property_length_rest - len
+                ]
+                (buffer body, buffer buf, async_handler_t func, this_type_sp self) mutable {
+                    props.emplace_back(
+                        v5::property::shared_subscription_available(body.begin(), body.end())
+                    );
+                    process_property_id(
+                        std::move(func),
+                        std::move(buf),
+                        rest,
+                        std::move(props),
+                        std::move(handler),
+                        std::move(self)
+                    );
+                },
+                std::move(self)
+            );
+        } break;
+        }
+    }
+
+    // process common
+
+    templat
```

Comment:
Not actually using anything captured by reference.

Reply:
I think that the code reference is something wrong. Could you update it?

Reply:
Yes. This comment went to the wrong place somehow...

I was pointing out that there are some lambda functions in one of the process_*** functions that are

`[&](){...}`

But don't actually use any of the variables in the local context. So the & can be removed.

Reply:
I will fix it.

Reply:
I noticed that `[&]` is for `this`. `version_` is referred int the lambda expression.
I know that it is better to write the capture list explicitly but in this case, the lambda expression is **defined and called in the same time**. Dangling reference can't happen. Only in this case, I think that `[&]` is acceptable. It is easy to maintain.

Reply:
Ok. Sounds good.